Complete challenge

kalebingerick opened the pull request.

Commit:
Tree class working and tested

Commit:
complete tests and tree.rb

jaybobo commented on lib/tree.rb:
```diff
@@ -1,39 +1,66 @@
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+class Tree
+  attr_accessor :height, :age, :apples, :alive
```
do all of these getters/setters need to be accessible?

jaybobo commented on spec/tree_spec.rb:
```diff
@@ -3,12 +3,73 @@
+  # as the tree ages, more apples grow.
+  # 2^age apples should be on the tree after it ages
+  it '.age!' do
+  	tree = Tree.new
```
take a look at let/let!
https://www.relishapp.com/rspec/rspec-core/v/2-5/docs/helper-methods/let-and-let

jaybobo commented on spec/tree_spec.rb:
```diff
@@ -3,12 +3,73 @@
+    expect(Tree.new.is_a? Tree).to be(true)
+  end
+
+  it '.initialize' do
```
we're pretty picky about indentation. keep an eye on this once you start at cmm

jaybobo commented on spec/tree_spec.rb:
```diff
@@ -3,12 +3,73 @@
+	expect(tree.apples == 8).to be true
+  end
+
+  it '.any_apples?' do
```
There's a lot going on in each of your `it` blocks. 
Clean these up with contexts... http://betterspecs.org/#contexts

paradime replied:
I don't really follow that the difference is or when contexts would be applicable to this example. I'm trying to follow arrange act assert here.

Commit:
added let statement and changed attribute accessor in tree

Commit:
Change apples to be an array,
Update tests to use more of rspec's built in functions (instead of just using expect(variable == integer).to be true)
Update add_apples to create a random diameter
Fix indentation

jaybobo commented:
:+1:
